File: scripts/generate_firebase_config.py
import os
import json

version = os.environ.get('VERSION', '1.0.0')
min_version = os.environ.get('MIN_VERSION', version)
updates = os.environ.get('UPDATES', '')

# Convert updates to markdown list
if updates:
    update_items = [item.strip() for item in updates.split('||') if item.strip()]
    release_notes = '\n'.join(f'- {item}' for item in update_items)
else:
    release_notes = ''

config = {
    "parameters": {
        "latest_version": {
            "defaultValue": {"value": version}
        },
        "min_version": {
            "defaultValue": {"value": min_version}
        },
        "release_notes": {
            "defaultValue": {"value": release_notes}
        }
    }
}

with open('firebase_release_config.json', 'w', encoding='utf-8') as f:
    json.dump(config, f, ensure_ascii=False, indent=2)

File: scripts/release_management/generate_firebase_config.js
const fs = require('fs');

// Get environment variables
const version = process.env.VERSION || '1.0.0';
const minVersion = process.env.MIN_VERSION || version;
const updates = process.env.UPDATES || '';

// Convert updates to markdown list
let releaseNotes = '';
if (updates) {
    const updateItems = updates.split('||')
        .map(item => item.trim())
        .filter(item => item);
    releaseNotes = updateItems.map(item => `- ${item}`).join('\n');
}

const config = {
    "parameters": {
        "latest_version": {
            "defaultValue": { "value": version }
        },
        "min_version": {
            "defaultValue": { "value": minVersion }
        },
        "release_notes": {
            "defaultValue": { "value": releaseNotes }
        }
    }
};

// Write config to file
fs.writeFileSync('firebase_release_config.json', JSON.stringify(config, null, 2), 'utf8');
console.log('Firebase release config generated successfully!');
